nextjs-app/google-oauth/app/login/google/route.ts

```diff
@@ -0,0 +1,27 @@
+import { cookies } from "next/headers";
+import { generateCodeVerifier, generateState } from "arctic";
+import { googleAuth } from "@/lib/auth";
+
+export async function GET(): Promise<Response> {
+  const state = generateState();
+  const codeVerifier = generateCodeVerifier();
+  const url = await googleAuth.createAuthorizationURL(state, codeVerifier, {
+    scopes: ["profile", "email"],
+  });
+
+  cookies().set("google_oauth_state", state, {
+    secure: true,
```

Comment:
this should be `secure: process.env.NODE_ENV === "production"` for localhost testing